org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/concurrent/WriteRequest.java

```diff
@@ -42,18 +42,23 @@ public void run() {
 		try {
 			previous.join();
 		} catch (Throwable t) {
-			// We are not interested in results, only to make sure that all previous requests are finished before running next write request.
+			if (!requestManager.isCancelException(t)) {
+				LOG.error("Error during request: ", t);
+			}
 		}
 		try {
 			U intermediateResult = this.nonCancellable.apply();
 			cancelIndicator.checkCanceled();
-			result.complete(cancellable.apply(cancelIndicator, intermediateResult));
+			V writeResult = cancellable.apply(cancelIndicator, intermediateResult);
+			complete(writeResult);
 		} catch (Throwable t) {
-			if (!requestManager.isCancelException(t)) {
-				LOG.error("Error during request: ", t);
-			}
-			result.completeExceptionally(t);
+			completeExceptionally(t);
```

Comment:
no logging here?

Reply:
completeExceptionally will log. I'll rename the method to make it more obvious.